docs/design-documents/hal/0004-pin-names-general-guidelines.md

```diff
@@ -0,0 +1,132 @@
+# General Pin Names design document
+
+# Table of contents
+
+1. [Table of contents](#table-of-contents).
+    1. [Revision history](#revision-history).
+1. [Introduction](#introduction).
+    1. [Overview and assumptions](#overview-and-assumptions)
+1. [Detailed design](#detailed-design).
+    1. [LEDs](#leds).
+    1. [Buttons](#buttons).
+1. [Non-valid definitions](#non-valid-definitions).
+1. [Testing compliance](#testing-compliance).
+
+
+### Revision history
+
+1.0 - Initial revision - Malavika Sajikumar, Marcelo Salazar - August 2020  
+This document is written for Mbed OS 6.
+
+## Introduction
+
+### Overview and assumptions
+
+Mbed OS is designed so that application code written in the platform is portable across different Mbed supported boards with the same hardware capabilities or interfaces. However, the code may not be truly portable due to the differences in pin name definitions for the same kind of interfaces across different boards. 
+
+This document provides specific guidelines and best practices for defining pin names that apply to boards that implement and are compliant wit Arduino Uno compliant connector.
+
```

Comment:
This requires change to explain that it's generic and applicable to LEDs/Buttons

Reply:
Maybe add here some comments that specific connector pins are described in some other specific page?
- Arduino Uno:  -add the link to the next page-
- Arduino Nano: todo?
- MikroBus: todo?

Reply:
Adding link to folder to indicate the availability of other docs. 
I'm not very keen on adding too much information on all connectors, but we can certainty add them in the future once we have a draft.